source/common/http/http2/codec_impl.cc

```diff
@@ -374,17 +422,53 @@ void ConnectionImpl::StreamImpl::readDisable(bool disable) {
 }
 
 void ConnectionImpl::StreamImpl::pendingRecvBufferHighWatermark() {
-  ENVOY_CONN_LOG(debug, "recv buffer over limit ", parent_.connection_);
-  ASSERT(!pending_receive_buffer_high_watermark_called_);
-  pending_receive_buffer_high_watermark_called_ = true;
-  readDisable(true);
+  if (!Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.defer_processing_backedup_streams")) {
+    ENVOY_CONN_LOG(debug, "recv buffer over limit ", parent_.connection_);
+    ASSERT(!pending_receive_buffer_high_watermark_called_);
+    pending_receive_buffer_high_watermark_called_ = true;
+    readDisable(true);
```

Comment:
wait, are we not read disabling in the new path? why?

Reply:
Prior it seems to never be called: https://github.com/envoyproxy/envoy/pull/19447/files/82147f0ce2cf7deff38beaa1af519dcc7cf9116d#diff-d36bad9503815c85f0a45bdd1b7342a73112349b6167e8170af677e0b9225990R329-R336

since we'd always eagerly pass the data through.  To preserve that behavior with deferred processing, and avoid getting stuck we disable it's watermarks.

Reply:
My worry is that after this change MarkDataConsumedForStream will be called early, effectively breaking flow control feedback.  Should you update the documentation of how H2 flow control works to reflect changes in this PR?

Reply:
I **think** the only case where this is called is when a filter adds more bytes to the received data in the decodeData()/decodeHeaders() call (i.e. decompressor filter). And I **think** the only outcome of this is reset stream.

I will look into this more, but I think whatever this callback is doing is largely a noop. I left a [comment](https://github.com/envoyproxy/envoy/blob/d3ecdce968830330fbffcfc492d28c212c509acf/source/common/http/http2/codec_impl.h#L324) a while ago, which is actually not quite correct, as it does not mention the edge case above.

Reply:
I think before we'd end up doing the "disable-h2-window-consume, dispatch data, enable-h2-window-consume" in the same callback usually. 

What are your thoughts on making this readDisable as follows: 
1) this is somehow not counted in the the pending data processing case e.g. to avoid getting stuck 
2) this is counted in the `MarkDataConsumed` case in `ConnectionImpl::OnData()` so we don't give the peer window 

This would thus use the overflow / low watermark to inform flow control, but it won't control whether we process received data